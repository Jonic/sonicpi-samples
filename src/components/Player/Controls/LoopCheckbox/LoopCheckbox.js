import './LoopCheckbox.scss'

import PropTypes from 'prop-types'
import React from 'react'

const LoopCheckbox = props => {
  const { isLoading, isLooping, loopCheckboxClickHandler } = props

  const handleClick = () => {
    loopCheckboxClickHandler()
  }

  return (
    <div className="c-loop-checkbox">
      Loop Sample:{' '}
      <input
        defaultChecked={isLooping}
        className="c-LoopCheckbox__input"
        disabled={isLoading}
        onClick={handleClick}
        type="checkbox"
      />
    </div>
  )
}

LoopCheckbox.propTypes = {
  isLoading: PropTypes.bool.isRequired,
  isLooping: PropTypes.bool.isRequired,
  loopCheckboxClickHandler: PropTypes.func.isRequired,
}

export default LoopCheckbox
