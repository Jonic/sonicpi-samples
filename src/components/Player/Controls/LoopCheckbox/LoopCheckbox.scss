@import '../../../../styles/component-helper';

.c-loop-checkbox {
}
