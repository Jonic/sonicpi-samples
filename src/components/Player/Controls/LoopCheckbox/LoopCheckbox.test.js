import LoopCheckbox from './LoopCheckbox'

const isLoading = false
const isLooping = false
const loopCheckboxClickHandler = jest.fn()

let wrapper

describe('<LoopCheckbox />', () => {
  beforeEach(() => {
    wrapper = shallow(
      <LoopCheckbox
        isLoading={isLoading}
        isLooping={isLooping}
        loopCheckboxClickHandler={loopCheckboxClickHandler}
      />,
    )
  })

  describe('Renders...', () => {
    it('without crashing', () => {
      expect(wrapper.find('.c-loop-checkbox')).toExist()
    })

    it('matches its snapshot', () => {
      expect(wrapper).toMatchSnapshot()
    })
  })
})
