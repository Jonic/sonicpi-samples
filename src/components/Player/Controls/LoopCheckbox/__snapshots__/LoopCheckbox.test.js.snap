// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoopCheckbox /> Renders... matches its snapshot 1`] = `
<div
  className="c-loop-checkbox"
>
  Loop Sample:
   
  <input
    className="c-LoopCheckbox__input"
    defaultChecked={false}
    disabled={false}
    onClick={[Function]}
    type="checkbox"
  />
</div>
`;
